Backend/WebApi-PointMap/Controllers/UserManagementController.cs

```diff
@@ -0,0 +1,228 @@
+                }
+            }
+        }
+
+        [HttpGet]
+        [Route("users/{managerId}")]
+        public IHttpActionResult GetUsersUnderManager(string managerId)
+        {
+            if (managerId == null)
+            {
+                return Content((HttpStatusCode)412, "Invalid payload.");
+            }
+            Guid ManagerId;
+            try
+            {
+                ManagerId = Guid.Parse(managerId);
+            }
+            catch (Exception)
+            {
+                return Content((HttpStatusCode)400, "Invalid SSO ID");
+            }
+            using (var _db = new DatabaseContext())
+            {
+                try
+                {
+                    var users = _db.Users
+                        .Where(u => u.Manager.Id == ManagerId)
+                        .Select(u => new {
+                            id = u.Id,
+                            username = u.Username,
+                            manager = u.ManagerId,
+                            city = u.City,
+                            state = u.State,
+                            country = u.Country,
+                            disabled = u.Disabled,
+                            isAdmin = u.IsAdministrator
+                        }).ToList();
+                    return Content((HttpStatusCode)200, users);
+                }
+                catch (Exception)
+                {
+                    return InternalServerError();
+                }
+            }
+        }
+
+        [HttpGet]
+        [Route("user/{token}")]
+        public IHttpActionResult GetUser(string token)
+        {
+            if (token == null || token == "")
+            {
+                return BadRequest("Invalid Session.");
+            }
+            using (var _db = new DatabaseContext())
+            {
+                SessionService _sessionService = new SessionService();
+                try
+                {
+                    var session = _sessionService.ValidateSession(_db, token);
+                    if (session == null)
+                    {
+                        return Content(HttpStatusCode.NotFound, "Session is no longer available.");
+                    }
+                    UserManagementManager _userManager = new UserManagementManager(_db);
+                    var user = _userManager.GetUser(session.UserId);
+                    return Ok(new
+                    {
+                        id = user.Id,
+                        username = user.Username,
+                        disabled = user.Disabled,
+                        isAdmin = user.IsAdministrator
+                    });
+                }
+                catch (Exception ex)
+                {
+                    return Content(HttpStatusCode.InternalServerError, ex);
+                }
+
+            }
+        }
+
+        [HttpDelete]
+        [Route("user/delete/{userId}")]
+        public IHttpActionResult DeleteUser(string userId)
```

Comment:
Every route should need to be authenticated with a token

Reply:
working on the rn